### PR TITLE
Persistent syntax highlight

### PR DESCRIPTION
**Pull request by raquel-ucl**
Rewrite of `syntax_highlight` so it covers error highlighting as well, so we can simplify the code a little bit and make error highlight persistent.
Also fixed the tooltip clearing functionality.

Still needs some refactoring, so please don't merge yet.

### CODE REVIEW

**Comment by raquel-ucl** [comment]
Ready to merge. 
Refactored Syntax/Error highlighting into a separate class. Error highlighting doesn't need separate functionality anymore, it's merged with syntax highlight.

File: python/nammu/controller/AtfAreaController.py [hunk 1]
```diff
@@ -18,20 +18,36 @@
...
         # Get a reference to the view's undo_manager
         self.undo_manager = self.view.undo_manager
+        # Initialise validation errors
+        self.validation_errors = {}
+        # Needed by syntax highlighter
+        self.edit_area_styledoc = self.edit_area.getStyledDocument()
+        self.line_numbers_styledoc = self.line_numbers_area.getStyledDocument()
+        # Syntax highlighting
+        self.syntax_highlighter = SyntaxHighlighter(self)
+#       self.syntax_highlighter = SyntaxHighlighter(self.line_numbers_styledoc)
```
Comment (jenshnielsen) [inline python/nammu/controller/AtfAreaController.py]: get rid of this line?

File: python/nammu/view/AtfEditArea.py [hunk 2]
```diff
@@ -60,6 +62,41 @@ def get_line_num(self, position):
         text = self.text[0:position]
         return text.count('\n') + 1
 
+    def setText(self, text):
+        '''
+        Override JTextPane's setText to call syntax highlighting so that it
+        still works when you set text from elsewhere in the code.
+        '''
+        super(AtfEditArea, self).setText(text)
```
Comment (jenshnielsen) [inline python/nammu/view/AtfEditArea.py]: Why does this not need a ```self.controller.update_line_numbers()```?

Comment (raquel-ucl) [inline python/nammu/view/AtfEditArea.py]: Added it now. IT was being called elsewhere everytime after setText, so it's better in here.

**Comment by raquel-ucl** [comment]
Jens found an error as error highlight persisted. Reproduce as:

1. Open `bb_2_008.atf`.
2. Lemmatise, see errors
3. Validate - returns not errors, tooltips are gone, but error styling stays.

This is now fixed. Syntax highlighting was only being invoked when there were errors coming from validation. Now it's invoked always after reloading the `validation_errors` dict.

**Pull request merged**